extension/src/popup/App.tsx

```diff
@@ -59,7 +66,10 @@ export function App() {
   return (
     <Provider store={store}>
       <GlobalStyle />
-      <Router />
+      <RouteWrapperEl>
+        <Router />
+      </RouteWrapperEl>
+      <Loading />
```

Comment:
keep the <Loading /> component behind the Router component so if the Route hasn't resolved, the loading screen will show

Reply:
Ah interesting, so this is like a backdrop that gets covered up if there isn't something else to show?

Reply:
yup, that's right